### DISCUSSION TITLE
Best practice to add fonts into a web component

### DISCUSSION THREAD

**Discussion by Alos** [open-wc/open-wc#2125] [Q&A] [answered]
I'm using: `npm init @open-wc` to create a brand new web component (not an application) called `my-foo` this component will use [mwc-icon-button](https://github.com/material-components/material-components-web-components/tree/master/packages/icon-button). The instructions say that I need to import the Material Icon font. What's the correct way of doing so using the generated code?

**Comment by bennypowers** [comment, marked as answer]
From https://www.npmjs.com/package/@material/mwc-icon:

```html
<link href="https://fonts.googleapis.com/css?family=Material+Icons&display=block" rel="stylesheet">

<mwc-icon>shopping_cart</mwc-icon>

<script type="module">
  import '@material/mwc-icon';
</script>
```

Just a `<link>` in the `<head>` of `index.html` is enough

**Reply by Alos** [reply]
There is no index.html generated by the command tho

**Reply by bennypowers** [reply]
so, either your users should add the link, or you can add one locally in your shadow root

**Reply by Alos** [reply]
I ended up doing something like this on the constructor:

```typescript
constructor() {
    super();
    const importNode = document.createElement('link');
    importNode.href =
      'https://fonts.googleapis.com/css?family=Material+Icons&display=block';
    importNode.rel = 'stylesheet';
    // Triggers import loading
    document.head.appendChild(importNode);
  }
```

But then I have to remember to remove it when the element is detached and making sure there are no dup links if the element is instantiated multiple times.

**Reply by bennypowers** [reply]
I think that's fine. I don't think you have to worry about double-loading or removing the link on disconnect, I tried that function in Chrome, Safari, and Firefox, and all of them only downloaded the resource once

You could always do 
```js
if (!document.head.querySelector('[href="https://fonts.googleapis.com/css?family=Material+Icons&display=block"]')) {
  /*..*/
}
```

**Reply by bennypowers** [reply]
or use `document.fonts` (might want to polyfill)

https://developer.mozilla.org/en-US/docs/Web/API/Document/fonts

**Reply by Alos** [reply]
Thanks! :)

**Reply by luas10c** [reply]
Before loading the fonts use to verify that the domains are accessible.
```html
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```

**Reply by erikomis** [reply]
like

**Reply by anandguptacv** [reply]
if (!document.head.querySelector('[href="https://fonts.googleapis.com/css?family=Material+Icons&display=block"]')) {
  /*..*/
}

**Comment by luas10c** [comment]
Before loading the fonts use to verify that the domains are accessible.
```html
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```

**Reply by erikomis** [reply]
fine

**Reply by anandguptacv** [reply]
if (!document.head.querySelector('[href="https://fonts.googleapis.com/css?family=Material+Icons&display=block"]')) {
  /*..*/
}

**Reply by abenkoula71** [reply]
hello

**Comment by erikomis** [comment]
fine

**Reply by luas10c** [reply]
fine!

**Reply by rivinduchamath** [reply]
fine

**Comment by Rofram** [comment]
very good

**Reply by luas10c** [reply]
fine!

**Reply by erikomis** [reply]
fine

**Comment by abenkoula71** [comment]
hello
